app/controllers/application_controller/buttons.rb

```diff
@@ -250,6 +251,17 @@ def applies_to_class_model(applies_to_class)
     applies_to_class == "ServiceTemplate" ? Service : applies_to_class.constantize
   end
 
+  def custom_button_done
+    url = SystemConsole.where(:vm => params[:id]).first.try(:url)
```

Comment:
@mkanoor : that sounds good. How do I do that? @gmcculloug suggested that I should pass an ID of some resource and work with that. Can you show me how I would do what you suggest?